views/default/page/components/module.php

```diff
@@ -11,43 +12,39 @@
  * @uses $vars['id']           Optional id for module
  * @uses $vars['show_inner']   Optional flag to leave out inner div (default: false)
  */
-
 $type = elgg_extract('type', $vars, false);
 $title = elgg_extract('title', $vars, '');
-$header = elgg_extract('header', $vars, '');
 $body = elgg_extract('body', $vars, '');
 $footer = elgg_extract('footer', $vars, '');
 $show_inner = elgg_extract('show_inner', $vars, false);
 
-$class = 'elgg-module';
-if ($type) {
-	$class = "$class elgg-module-$type";
-}
-$additional_class = elgg_extract('class', $vars, '');
-if ($additional_class) {
-	$class = "$class $additional_class";
-}
+$attrs = [
+	'id' => elgg_extract('id', $vars),
+	'class' => (array) elgg_extract('class', $vars, []),
+];
 
-$id = '';
-if (isset($vars['id'])) {
-	$id = "id=\"{$vars['id']}\"";
+$attrs['class'][] = 'elgg-module';
+if ($type) {
+	$attrs['class'][] = "elgg-module-$type";
 }
 
 if (isset($vars['header'])) {
-	$header = "<div class=\"elgg-head\">$header</div>";
-} elseif ($title) {
-	$header = "<div class=\"elgg-head\"><h3>$title</h3></div>";
+	$header = elgg_extract('header', $vars, '');
+} else if ($title) {
+	$header = elgg_format_element('h3', [], $title);
 }
 
-$body = "<div class=\"elgg-body\">$body</div>";
-
+if (isset($header)) {
```

Comment:
This looks right. Only thing bugs me is that this relies on logic where a variable may or may not be present, and I think that invites errors. I'd probably prefer setting it to null if missing and explicitly checking null later.